misc.sk

```diff
@@ -50,19 +50,34 @@ random items:
 	sugar = minecraft:sugar
 	ender pearl¦s = minecraft:ender_pearl
 	blaze rod¦s = minecraft:blaze_rod
-	nether wart = minecraft:nether_wart
+	nether wart¦s = minecraft:nether_wart
 	(ender eye¦s|eye¦s of ender) = minecraft:ender_eye
 	(bottle¦s o' enchanting|xp bottle¦s|experience bottle¦s) = minecraft:experience_bottle
 	fire charge¦s = minecraft:fire_charge
 	(book and quill¦s|writable book¦s) = minecraft:writable_book
-	# TODO maps?
+	[empty] map = minecraft:map
+	filled map = minecraft:filled_map
 	nether star¦s = minecraft:nether_star
-	# TODO fireworks
 	nether brick¦s = minecraft:nether_brick
 	nether quartz = minecraft:quartz
 	prismarine shard¦s = minecraft:prismarine_shard
 	prismarine crystal¦s = minecraft:prismarine_crystals
 	rabbit hide¦s = minecraft:rabbit_hide
+	
+{firework types}:
+	(0|zero) = {Fireworks:{Flight:0}}
+	(1|one) =  {Fireworks:{Flight:1}}
+	(2|two) = - {Fireworks:{Flight:2}}
+	(3|three) = - {Fireworks:{Flight:3}}
+	
+before flattening:
+	minecraft version = 1.12.2 or older
+	(firework|firework rocket)¦s = fireworks
+	[(flight|duration|flight duration) {firework types} ] (firework|firework rocket)¦s = fireworks
+	
+after flattening:
+	minecraft version = 1.13 or newer
+	[(flight|duration|flight duration) {firework types} ] (firework|firework rocket)¦s = firework_rocket
```

Comment:
The two instances of `(firework|firework rocket)` could be changed to `(firework [rocket])`.

Also, there's an extra space in the first optional part after `{firework types}`. Not a huge deal obviously but while you're making fixes it could be removed easily.